packages/nextjs/src/config/withSentryConfig.ts

```diff
@@ -5,25 +5,44 @@ import type {
   NextConfigObject,
   NextConfigObjectWithSentry,
   SentryWebpackPluginOptions,
+  UserSentryOptions,
 } from './types';
 
 /**
  * Add Sentry options to the config to be exported from the user's `next.config.js` file.
  *
  * @param exportedUserNextConfig The existing config to be exported prior to adding Sentry
  * @param userSentryWebpackPluginOptions Configuration for SentryWebpackPlugin
+ * @param sentryWizardAddon Addons from automatic config.next.js merging from Sentry Wizard
  * @returns The modified config to be exported
  */
 export function withSentryConfig(
   exportedUserNextConfig: ExportedNextConfig = {},
   userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+  sentryWizardAddon?: UserSentryOptions,
 ): NextConfigFunction | NextConfigObject {
   return function (phase: string, defaults: { defaultConfig: NextConfigObject }): NextConfigObject {
-    if (typeof exportedUserNextConfig === 'function') {
-      const userNextConfigObject = exportedUserNextConfig(phase, defaults);
+    // If there are addons from the Wizard, add them to existing config
+    if (sentryWizardAddon) {
+      let userNextConfigObject;
+      if (typeof exportedUserNextConfig === 'function') {
+        userNextConfigObject = exportedUserNextConfig(phase, defaults);
+      } else {
+        userNextConfigObject = exportedUserNextConfig;
+      }
+      if (userNextConfigObject.sentry) {
```

Comment:
@roggenkemper I don't think you need this, can just use `userNextConfigObject.sentry = { ...userNextConfigObject.sentry, ...sentryWizardAddon }` even if `userNextConfigObject.sentry,` is undefined